### PR TITLE
Using USWDS styles for input fields and labels on TI forms

### LINKED ISSUE #6912: Update TI client edit and add pages with latest UX design
#### Describe the solution presented in this issue.
The TI client edit and new client views differ from the latest TI dashboard mocks.  In this issue, we'll make them match as much as possible.  All input fields should use the USWDS input component.  The back link should have an arrow icon.  There shouldn't be a border around the form.

#### Link to Figma mocks: 
[mocks](https://www.figma.com/file/yfhnUmE2FsonLtAqfeEfoo/TI-Edit?type=design&node-id=1112%3A1348&mode=design&t=y9ukT6E6JXHgHNZ8-1)

#### Add screenshot of relevant mocks for this issue here: 

image

#### Additional Context
Add any other context, link related issues, etc. 

#### Done when:
- [ ] All input fields use USWDS input component
- [ ] Labels are correct and internationalized
- [ ] No border around the form
- [ ] Back link has arrow icon
- [ ] 'Cancel' button has white background
- [ ] Buttons use USWDS styles

### PR DESCRIPTION
**Pull request by melanie-exygy**
### Description

Using USWDS styles for input fields and labels on the TI add/edit client form.  Since these fields are of type `FieldWithLabel`, some refactoring was needed to move CSS class assignment further down the call stack and to separate it from attribute assignment and validation error creation.  Much of the `FieldWithLabel` logic was duplicated with the only change being USWDS classes.  This duplication is temporary given that we plan to migrate all CF input fields to USWDS with the North Star work.

### Instructions for manual testing

1.  Log in as a Trusted Intermediary.
2. Click on the "Add new client" button.
3. Notice that the input fields now have USWDS styles.
4. Click "Save".
5. Notice that validation error messages still show, with USWDS error styles.

image

### Issue(s) this completes

Fixes #6912

### CODE REVIEW

File: browser-test/image_snapshots/end_to_end_enumerators_test/enumerator-field.png
Comment (melanie-exygy) [inline browser-test/image_snapshots/end_to_end_enumerators_test/enumerator-field.png:1]: The difference here and in some of the other snapshots is that the label with text "Administrative identifier..." now has the same styles as other labels (it isn't more bold and the letters aren't closer together).  Doing this means that the text is a bit more spaced out and that makes the tooltip icon wrap onto the next line.  I don't think it looks great, but it only looks this way with this particular screen size.  Also, the wrapping behavior is unchanged from what it was before.  Even with `display:block` the label still wraps when the screen size is narrower (you can verify this on staging).

File: server/app/views/applicant/EditTiClientView.java [hunk 1]
```diff
@@ -310,13 +310,13 @@ private DivTag renderAddOrEditClientForm(
...
                 .with(
-                    firstNameField.getInputTag(),
-                    middleNameField.getInputTag(),
-                    lastNameField.getInputTag(),
-                    phoneNumberField.getInputTag(),
-                    emailField.getEmailTag(),
-                    dateOfBirthField.getDateTag(),
-                    tiNoteField.getTextareaTag(),
+                    firstNameField.getUSWDSInputTag(),
+                    middleNameField.getUSWDSInputTag(),
```
Comment (swatkat1) [inline server/app/views/applicant/EditTiClientView.java:314]: This is cleaver and a lot more understandable.

File: server/app/views/components/FieldWithLabel.java [hunk 2]
```diff
@@ -428,7 +428,19 @@ private DivTag getNonNumberInputTag() {
     } else {
       throw new RuntimeException("non-number tag expected");
     }
-    return applyAttrsAndGenLabel(inputFieldTag);
+    return applyAttrsClassesAndLabel(inputFieldTag);
+  }
+
+  private DivTag getUSWDSNonNumberInputTag() {
```
Comment (swatkat1) [inline server/app/views/components/FieldWithLabel.java:434]: This is doing almost the same things as getNonNumberInputTag(), can you combine the two? May be pass in a variable or set a class param to identify it as a USWDS component.

File: server/app/views/components/FieldWithLabel.java [hunk 3]
```diff
@@ -443,7 +455,24 @@ public DivTag getTextareaTag() {
...
         textareaFieldTag.withCols(String.valueOf(this.cols.getAsLong()));
       }
-      return applyAttrsAndGenLabel(textareaFieldTag);
+      return applyAttrsClassesAndLabel(textareaFieldTag);
+    }
+
+    throw new RuntimeException("needs to be textarea tag");
+  }
+
+  public DivTag getUSWDSTextareaTag() {
```
Comment (swatkat1) [inline server/app/views/components/FieldWithLabel.java:464]: This is also doing almost the same things as getTextareaTag(). Can you look to combining them?

Comment (melanie-exygy) [reply server/app/views/components/FieldWithLabel.java:464]: I see what you're saying.  I considered the option of passing in a `isUSWDS` variable here, but there are 21 usages of this method throughout the app and I'd have to add the parameter in all of those places.  Then, when USWDS becomes the norm for our input fields, we'd have to remove the parameters in all those places.  By duplicating the function, we temporarily have two almost identical functions, but we'll have less to change when we are only using USWDS for input fields.  We'll be able to blend the two functions into one at that time.  Does that sound reasonable?

File: server/app/views/components/FieldWithLabel.java (hunk 2 above)
Comment (melanie-exygy) [reply server/app/views/components/FieldWithLabel.java]: Done

**Review by swatkat1 — APPROVED** [review]
Nicely done!!!

File: server/app/views/components/FieldWithLabel.java (hunk 3 above)
Comment (swatkat1) [reply server/app/views/components/FieldWithLabel.java:454]: ok sounds good! - Can you add a TODO item to make sure we followup. Thank you.

Comment (melanie-exygy) [reply server/app/views/components/FieldWithLabel.java:454]: Good idea, will do!

File: server/app/views/components/FieldWithLabel.java [hunk 4]
```diff
@@ -428,7 +428,19 @@ private DivTag getNonNumberInputTag() {
...
+    applyAttributesFromMap(inputFieldTag);
+    if (!this.fieldType.equals("number")) {
+      inputFieldTag.withValue(this.fieldValue);
+    } else {
+      throw new RuntimeException("non-number tag expected");
+    }
+    return applyUSWDSAttrsClassesAndLabel(inputFieldTag);
   }
 
   /** Public final tag getters * */
```
[the remark this answers is not in the document]
Comment (melanie-exygy) [reply server/app/views/components/FieldWithLabel.java:436]: I didn't add this comment.  I think it's just marking the start of the public tag getter methods (vs the internal methods).

File: server/app/views/components/FieldWithLabel.java (hunk 3 above)
Comment (melanie-exygy) [reply server/app/views/components/FieldWithLabel.java:456]: Done

**Pull request merged**